cpp/src/arrow/CMakeLists.txt

```diff
@@ -793,10 +777,25 @@ if(ARROW_COMPUTE)
        compute/kernels/vector_replace.cc
        compute/kernels/vector_run_end_encode.cc
        compute/kernels/vector_select_k.cc
-       compute/kernels/vector_sort.cc)
+       compute/kernels/vector_sort.cc
+       compute/key_hash_internal.cc
+       compute/key_map_internal.cc
+       compute/light_array_internal.cc
+       compute/row/encode_internal.cc
+       compute/row/compare_internal.cc
+       compute/row/grouper.cc
+       compute/row/row_encoder_internal.cc
+       compute/row/row_internal.cc
+       compute/util.cc
+       compute/util_internal.cc)
```

Comment:
Could you add a description why we should move them to the `ARROW_COMPUTE` block to the PR description?